src/Common/src/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -415,5 +415,27 @@ public override PInvokeMetadata GetPInvokeMethodMetadata()
 
             return new PInvokeMetadata(moduleName, name, (PInvokeAttributes)import.Attributes);
         }
+
+        public override PInvokeArgumentMetadata[] GetPInvokeArgumentMetadata()
+        {
+            if (!IsPInvoke)
+                return default(PInvokeArgumentMetadata[]);
+
+            MetadataReader metadataReader = MetadataReader;
+
+            ParameterHandleCollection parameterHandles = metadataReader.GetMethodDefinition(_handle).GetParameters();
+            PInvokeArgumentMetadata[] argumentMetadata = new PInvokeArgumentMetadata[parameterHandles.Count];
```

Comment:
We add caches only once we can prove that they improve something significant. Short term garbage like this is very cheap, and it should be typically needed exactly once as you have said, so it is unlikely that it will become a problem.